nucleof072rb/Core/Src/main.c

```diff
@@ -1,181 +1,219 @@
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+
+#include <stdint.h>
+
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+
+/* USER CODE BEGIN PV */
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+
+  /* USER CODE BEGIN 1 */
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  /* USER CODE BEGIN 2 */
+
+  const uint16_t counter_period = 60000;
+
+  // ADC receives one byte at a time: start bit, input channel configuration (0x80 = single-ended input mode and read from CH0), don't care
+  uint8_t tx[3] = {0x1, 0x80, 0x0};
+  uint8_t rx[3];
+
+  // Push CS high (ADC active low)
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+  // Start MCU timer
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  while (1)
+  {
+    /* USER CODE END WHILE */
+
+	  // Pull chip select low to begin communication with ADC
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  // Send and receive data from ADC60000
+	  HAL_SPI_TransmitReceive(&hspi1, tx, rx, sizeof(tx), 0); // What goes in timeout?
+
+	  // Push chip select high to end communication with ADC
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+	  // First byte is unknown. The last two bits of the second byte appended with all bits of the third byte is the conversion result
+	  const uint16_t conv = (((uint16_t)rx[1] & 0x03) << 8) | ((uint16_t)rx[2] & 0xFF);
+
+	  // Result of 0 means 5% duty cycle (1ms on), 1 << 10 - 1 means 10% duty cycle (2ms on)
+	  const uint16_t duty_cycle = 0.05 * (1 + (float)conv / ((1 << 10) - 1));

```

Comment:
This calculation works but in my opinion it is way more readable to represent 1023 just as `1023` rather than `(1 << 10) - 1)`. It gets rid of some bloat and makes it easier for someone reviewing your code to understand what you are doing. Also, make sure you don't use magic numbers and create consts for these or #define